index.js

```diff
@@ -16,37 +16,67 @@ app.listen(PORT, function(req, res){
 	console.log(`Server is running @ ${PORT}`);
 })
 
+//End Date Func
+var date = new Date();
+var end = format(date);
+
+// middleware to get date
+var startDate = function(req,res,next){
+	axios.get(reqUrl,{ params:{
+		query: req.query.query,
+		sort: 'P_PDATE_D asc',
+		pageSize:1,
+		synonym: true,
+		format: 'json'  }})
+	.then(function(response){
+		 req.start = response.data.resultList.result[0].firstPublicationDate;
+		 next();		
+	})
+	.catch(function(err){
+	 	res.status(500).json({
+		 	content:'Api request failed'
+		})	
+	})
+}
+app.use(startDate);
 app.get('/getWithYear', malaria);
 
 function malaria(req, res){	
-		axios.get(reqUrl,{ params:{query: req.query.query,
-			date: `( FIRST_PDATE:[${req.query.start} TO ${req.query.end}])`,
-			sort: 'CITED desc',
-			pageSize:100,
-			synonym: true,
-			format: 'json' }})
-		.then(function(response){
-			var rest = {};
+	axios.get(reqUrl,{ params:{query: req.query.query,
+		date:`( FIRST_PDATE:[${req.query.start || req.start} TO ${req.query.end || end}])`,
+		sort: 'CITED desc',
+		pageSize:10,
+		synonym: true,
+		format: 'json' }})
+	.then(function(response){
+		var rest = {};
+		if(response && typeof response.data != 'undefined' && typeof response.data.resultList !='undefined' && response.data.resultList.result ) {
 			const data = response.data.resultList.result;	
 			data.forEach(function(result) {
 				var key = result.pubYear;				
-  				var responseData = {title: result.title, author: result.authorString, year: result.pubYear, count: result.citedByCount};
-  				if(rest.hasOwnProperty(key)){
-  					rest[key].push(responseData);
-  				} else {
-  					rest[key] = [];
-  					rest[key].push(responseData);
-  				}  	 			 				
-			});	
-			res.status(200).json(rest); 		
-		})
-		.catch(function(err){
-			 res.status(500).json({
-			 	content:'Api request failed'
-			 })	
-		})
-	}
-			
-
-
+				var responseData = {title: result.title, author: result.authorString, year: result.pubYear, count: result.citedByCount};
+				if(rest.hasOwnProperty(key)){
+					rest[key].push(responseData);
+				} else {
+					rest[key] = [];
+					rest[key].push(responseData);
+				}  	 			 				
+			});
+			console.log(response)	
+			res.status(200).json(rest); 
+		}		
+	})
+	.catch(function(err){
+		res.status(500).json({
+		 	content:'Api request failed'
+		})	
+	})
+}
+//date format function
 
+function format(date) {
```

Comment:
@Firnaa5 Very nice. :clap:  But create a new file called `utils.js` that is where you hold generic utilities like this.